### DISCUSSION TITLE
VS Code 上でのテスト実行時における文字化けの対処について

### DISCUSSION THREAD

**Discussion by rnakagawa16** [AlesInfiny/maia#1311] [05: Q&A] [answered]
# 概要

私の環境では、VS Code 上でテストを実行する際、文字化けが発生していました。
この際、ブラウザ上で上位に出てくる対処法を実施することで文字化けを回避していましたが、
この対処法だと npm スクリプトが正常に作動しないことが判明したため、他の方法を試す必要があると判断しました。

# これまでの対処法

私自身では、VS Code の settings.json に設定を記載することで対応していました。

参考: https://kumipuro.tech/vscode_garbled_text/

しかし、この方法を使用すると npm スクリプト上で動作させる際に以下のページに示すようなエラーが発生することが判明しました。

参考: https://drroot.page/wp/?p=216

よってほかの方法で文字化けを解消できないか調査したところ、以下の方法が有用でした。

参考: https://pystyle.info/vscode-change-default-encoding-of-terminal-to-utf8/

# 質問事項

1. 皆さんはテスト実行時など、VS Code のターミナル上で文字化けが発生するような環境でしたか？
2. 文字化けが発生していた場合、どのような対応をとっていましたか？
3. この内容をローカル環境構築の説明に追記すべきだと思いますか？

**Comment by KentaHizume** [comment]
あまり参考にならない回答で申し訳ないのですが、私はVSCodeのターミナル上での文字化けにはあまり出会っておりません。
私の場合は基本的にターミナルとしてGit Bashを使っており、ここでの文字コードがUTF8になっているためかと思います。

```terminal
 locale
LANG=ja_JP.UTF-8
LC_CTYPE="ja_JP.UTF-8"
LC_NUMERIC="ja_JP.UTF-8"
LC_TIME="ja_JP.UTF-8"
LC_COLLATE="ja_JP.UTF-8"
LC_MONETARY="ja_JP.UTF-8"
LC_MESSAGES="ja_JP.UTF-8"
LC_ALL=
```

![image](https://github.com/user-attachments/assets/6697124e-fa57-41c1-8bdc-a3fef9ada56e)

**Reply by KentaHizume** [reply]
申し訳ありません、確認したところGit Bashでも文字化けが発生いたしました。

```terminal
> Task :web:test

HealthCheckTest > testGet_04_繝倥Ν繧ｹ繝√ぉ繝・け_DB豁｣蟶ｸ蜍穂ｽ懃｢ｺ隱・FAILED
    java.lang.AssertionError at HealthCheckTest.java:40

14 tests completed, 1 failed
```

**Comment by rnakagawa16** [comment, marked as answer]
1. 皆さんはテスト実行時など、VS Code のターミナル上で文字化けが発生するような環境でしたか？
　メンバーで共通で起きている事象であることがわかりました。
2. 文字化けが発生していた場合、どのような対応をとっていましたか？
　https://pystyle.info/vscode-change-default-encoding-of-terminal-to-utf8/
　上記の方法をほかのメンバーもとっていることがわかりました。
4. この内容をローカル環境構築の説明に追記すべきだと思いますか？
　この方法はDiscussion上のみで展開するものとし、ドキュメントとしては記載しない方針といたします。

ご協力いただいた皆様ありがとうございました。
